File: Server/getBlockedBuddies.php
<?php
require "globalFunctions.php";
$pdo = new PDO("pgsql:host=" . HOST . ";port=" . PORT . ";dbname=" . DATABASE . ";user=".USERNAME . ";password=" . PASSWORD);

$myId = $_POST["myId"];
$token = $_POST["token"];
$bottomBlockedAt = $_POST["bottomBlockedAt"];

$isValid =
    isAuthenticated($myId, $token) &&
    isValidTime($bottomBlockedAt);
if (!$isValid) {
    $pdo = null;
    die("Invalid");
}

$query = "
select account.user_id,
       account.username,
       account.birthday,
       account.gender,
       account.country,
       account.intro,
       account.last_visited_at,
       block.blocked_at
from   block
       left join account
              on block.buddy_id = account.user_id
where  account.deleted_at is null
       and block.user_id = ?
       and block.blocked_at < ?
order  by block.blocked_at desc
limit  20;";
$stmt = $pdo->prepare($query);
$stmt->execute(array($myId, $bottomBlockedAt));

$counter = 1; // PHP arrays start from 1
$return = array();
while ($row = $stmt->fetch()) {
    $return[$counter++] = array(
        "buddyId" => $row[0],
        "username" => $row[1],
        "birthday" => $row[2],
        "gender" => $row[3],
        "country" => $row[4],
        "intro" => $row[5],
        "lastVisitedAt" => $row[6],
    );

    $bottomBlockedAt = $row[7];
}

$return[$counter++] = array(
    "bottomBlockedAt" => $bottomBlockedAt,
);

echo json_encode($return);

$pdo = null;

File: Server/getBlockedBuddies.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"net/http"
	"time"
)

func getBlockedBuddies(w http.ResponseWriter, r *http.Request) {
	r.ParseForm()
	token := r.Form.Get("token")
	bottomBlockedAt := r.Form.Get("bottomBlockedAt")
	myId := getMyId(token)

	isValid :=
		isAuthenticated(myId) &&
			isValidTime(bottomBlockedAt)
	if !isValid {
		fmt.Fprintln(w, "Invalid")
		return
	}

	query := `
	select account.user_id,
	       account.username,
	       account.gender,
	       account.birthday,
	       account.country,
	       account.intro,
	       account.last_visited_at,
	       block.blocked_at
	from   block
	       left join account
	              on block.buddy_id = account.user_id
	where  account.deleted_at is null
	       and block.user_id = $1
	       and block.blocked_at < $2
	order  by block.blocked_at desc
	limit  20;`

	rows, err := pool.Query(context.Background(), query, myId, bottomBlockedAt)
	handleError(err)
	defer rows.Close()

	mapper := map[int]interface{}{}
	newBottomBlockedAt := toTime(bottomBlockedAt)

	for rows.Next() {
		var buddyId string
		var username string
		var gender int
		var birthday time.Time
		var country int
		var intro string
		var lastVisitedAt time.Time
		var blockedAt time.Time

		err = rows.Scan(&buddyId, &username, &gender, &birthday, &country, &intro, &lastVisitedAt, &blockedAt)
		handleError(err)

		newBottomBlockedAt = blockedAt

		newMapper := map[string]interface{}{
			"buddyId":       buddyId,
			"username":      username,
			"gender":        gender,
			"birthday":      birthday,
			"country":       country,
			"intro":         intro,
			"lastVisitedAt": lastVisitedAt,
		}
		mapper[len(mapper)] = newMapper
	}
	handleError(rows.Err())

	newMapper := map[string]interface{}{
		"bottomBlockedAt": newBottomBlockedAt,
	}
	mapper[len(mapper)] = newMapper

	json, err := json.Marshal(mapper)
	handleError(err)

	w.Header().Set("Content-Type", "application/json")
	w.Write(json)
}
